File: sort.go
package main

import (
	"math/rand"
	"time"
)

const POPULATION = 100

func Qsort(data []uint32) {
	size := len(data)
	if size > 1 {
		pivot := data[0]
		lesser := make([]uint32, size)
		greater := make([]uint32, size)
		lesser_index := 0
		greater_index := 0
		for _, value := range data[1:] {
			if value < pivot {
				lesser[lesser_index] = value
				lesser_index++
			} else {
				greater[greater_index] = value
				greater_index++
			}
		}
		lesser = lesser[:lesser_index]
		greater = greater[:greater_index]
		Qsort(lesser)
		Qsort(greater)
		i := 0
		for _, value := range lesser {
			data[i] = value
			i++
		}
		data[i] = pivot
		i++
		for _, value := range greater {
			data[i] = value
			i++
		}
	}
}

func main() {
	// Setup
	rand.Seed(time.Now().UnixNano())
	data := make([]uint32, POPULATION)
	for i := 0; i < POPULATION; i++ {
		data[i] = uint32(rand.Intn(100))
	}

	// Process
	Qsort(data)

	// Output
	for _, value := range data {
		print(value, "\t")
	}
	print("\n")
}

File: sort.c
#include <stdlib.h>
#include <stdio.h>
#include <time.h>

#define POPULATION 100

static int cmpint(const void*, const void*);

int main() {
    /* Setup */
    time_t t;
    time(&t);
    srand((unsigned int) t);
    u_int32_t data[POPULATION];
    for (size_t i = 0; i < POPULATION; ++i)
        data[i] = (u_int32_t) rand() % 100;

    /* Process */
    qsort(
        data,
        POPULATION,
        sizeof(u_int32_t),
        (int (*)(const void *, const void *)) cmpint
    );

    /* Output */
    for (size_t i = 0; i < POPULATION; ++i)
        printf("%u\t", data[i]);
    printf("\n");

    return EXIT_SUCCESS;
}

int cmpint(const void *a, const void *b) {
    return *(u_int32_t *) a - *(u_int32_t *) b;
}
